### DISCUSSION TITLE
how to create client kubeconfig？

### DISCUSSION THREAD

**Discussion by love-miao** [k3s-io/k3s#7017] [Q&A]
I want to generate the kubeconfig file used by the client through/var/lib/ranger/k3s/server/tls/server-ca. key | crt, and report an error after the generation: error: You must be logged in to the server (Unauthorized), but there is a problem with native kubernets. Is there any restriction on k3s?

The process is as follows：
`openssl genrsa -out ops-user.key 2048

openssl req -new -key ops-user.key -subj “/CN=ops-user/O=ops” -out ops-user.csr

openssl x509 -req -in ops-user.csr
-CA /var/lib/rancher/k3s/server/tls/server-ca.crt
-CAkey /var/lib/rancher/k3s/server/tls/server-ca.key
-CAcreateserial
-out ops-user.crt -days 3650

export KUBE_APISERVER=“https://192.168.100.100:6443”
kubectl config set-cluster kubernets
–certificate-authority=/var/lib/rancher/k3s/server/tls/server-ca.crt
–server=${KUBE_APISERVER}
–embed-certs=true
–kubeconfig=ops-user.yaml

kubectl config set-credentials ops-user
–client-certificate=ops-user.crt
–client-key=ops-user.key
–embed-certs=true
–kubeconfig=ops-user.yaml

kubectl config set-context kubernets
–cluster=kubernets
–user=ops-user
–kubeconfig=ops-user.yaml

kubectl config use-context kubernets --kubeconfig=ops-user.yaml

kubectl apply -f k8s_create_kubeconfig_ClusterRoleUser.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
name: ops-user
rules:

apiGroups:
‘*’
resources:
‘*’
verbs:
‘*’
nonResourceURLs:
‘*’
verbs:
‘*’
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
name: ops-user
roleRef:
apiGroup: rbac.authorization.k8s.io
kind: ClusterRole
name: ops-user
subjects:

apiGroup: rbac.authorization.k8s.io
kind: User
name: ops-user
`

**Comment by brandond** [comment]
You're trying to generate a client cert with the server CA. Use the client CA.

**Comment by brandond** [comment]
The proper way to do this would be to use the built in certificate signing workflow though:
https://github.com/k3s-io/k3s/discussions/6689#discussioncomment-4990056
